File: app/src/main/java/com/gft/testegft/network/ApiInterface.java
package com.gft.testegft.network;

import com.gft.testegft.login.data.LoginResponse;

import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ApiInterface {

    @FormUrlEncoded
    @POST("login/")
    Single<LoginResponse> login(@Field("user") String username,
                                @Field("password") String password);

    @GET("statements/{id}")
    Single<Response> getStatements(@Path("id") String id);
}

File: app/src/main/java/com/earaujo/santander/repository/rest/ApiInterface.kt
package com.earaujo.santander.repository.rest

import com.earaujo.santander.repository.models.LoginResponse
import com.earaujo.santander.repository.models.StatementsResponse
import retrofit2.Call
import retrofit2.http.*

interface ApiInterface {

    @Headers("Content-Type: application/x-www-form-urlencoded")
    @FormUrlEncoded
    @POST("login")
    fun login(
        @Field("user") user: String,
        @Field("password") password: String
    ): Call<LoginResponse>

    @Headers("Content-Type: application/x-www-form-urlencoded")
    @GET("statements/1")
    fun statements(
    ): Call<StatementsResponse>

}